src/Kafka/ManageKafkaPermissions/components/SelectAccount.tsx

```diff
@@ -0,0 +1,187 @@
+import React, { useState } from "react";
+import { useTranslation } from "react-i18next";
+import { FormGroupWithPopover } from "../../../shared/FormGroupWithPopover";
+import { Principal, PrincipalType } from "./ManagePermissionsModal";
+import {
+  Divider,
+  Select,
+  SelectGroup,
+  SelectOption,
+  SelectVariant,
+  ValidatedOptions,
+  SelectProps,
+  SelectOptionObject,
+} from "@patternfly/react-core";
+
+export type Validated<T> = {
+  value: T;
+  validated?: ValidatedOptions;
+  errorMessage?: string;
+};
+
+export type SelectAccountProps = {
+  id: Validated<string | undefined | SelectOptionObject>;
+  onChangeAccount: React.Dispatch<
+    React.SetStateAction<Validated<string | undefined | SelectOptionObject>>
+  >;
+  initialOptions: Principal[];
+  onEscapeModal: (closes: boolean) => void;
+};
+
+export const SelectAccount: React.VFC<SelectAccountProps> = ({
+  onChangeAccount,
+  id,
+  initialOptions,
+  onEscapeModal,
+}) => {
+  const { t } = useTranslation(["manage-kafka-permissions"]);
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+
+  const onToggle = (newState: boolean | ((prevState: boolean) => boolean)) => {
+    if (newState) {
+      onEscapeModal(false);
+    } else onEscapeModal(true);
+
+    setIsOpen(newState);
+  };
+
+  const clearSelection = () => {
+    onChangeAccount({
+      value: undefined,
+      validated: ValidatedOptions.error,
+      errorMessage: t("common:required"),
+    });
+    setIsOpen(false);
+  };
+
+  const onSelect: SelectProps["onSelect"] = (_, selection) => {
+    onChangeAccount({
+      value: selection,
+      validated: ValidatedOptions.default,
+    });
+    setIsOpen(false);
+  };
+  const options = initialOptions.length
+    ? [
+        <SelectGroup key="all_accounts_group">
+          <SelectOption
+            key="*"
+            value="*"
+            description={t(
+              "manage_permissions_dialog.all_accounts_description"
+            )}
+          >
+            {t("manage_permissions_dialog.all_accounts_title")}
+          </SelectOption>
+        </SelectGroup>,
+        <Divider key="all_accounts_divider" />,
+        <SelectGroup
+          label={t(
+            "manage_permissions_dialog.all_accounts_service_account_group"
+          )}
+          key="service_accounts_group"
+        >
+          {initialOptions
```

Comment:
It's a good practice to keep the logic on the jsx at a minimum to make it easier to read. This filtering and sorting should be done before the render, and the jsx moved inside the return of the render.